src/Text/Markdown/SlamDown/Pretty.purs

```diff
@@ -121,28 +123,28 @@ prettyPrintTextBoxValue t =
     SD.Time prec (Identity def) → prettyPrintTime prec def
     SD.DateTime prec (Identity def) → prettyPrintDateTime prec def
 
-prettyPrintDate ∷ SD.DateValue → String
-prettyPrintDate { day, month, year } =
-  printIntPadded 4 year
+prettyPrintDate ∷ DT.Date → String
+prettyPrintDate d =
+  printIntPadded 4 (fromEnum $ DT.year d)
     <> "-"
-    <> printIntPadded 2 month
+    <> printIntPadded 2 (fromEnum $ DT.month d)
     <> "-"
-    <> printIntPadded 2 day
+    <> printIntPadded 2 (fromEnum $ DT.day d)
 
-prettyPrintTime ∷ SD.TimePrecision → SD.TimeValue → String
-prettyPrintTime prec { hours, minutes, seconds }=
-  printIntPadded 2 hours
+prettyPrintTime ∷ SD.TimePrecision → DT.Time → String
+prettyPrintTime prec t =
+  printIntPadded 2 (fromEnum $ DT.hour t)
     <> ":"
-    <> printIntPadded 2 minutes
+    <> printIntPadded 2 (fromEnum $ DT.minute t)
     <> case prec of
-        SD.Seconds -> ":" <> printIntPadded 2 (M.fromMaybe 0 seconds)
+        SD.Seconds -> ":" <> printIntPadded 2 (fromEnum $ DT.second t)
         _ -> ""
 
-prettyPrintDateTime ∷ SD.TimePrecision → SD.DateTimeValue → String
-prettyPrintDateTime prec { date, time } =
-  prettyPrintDate date
-    <> "T"
-    <> prettyPrintTime prec time
+prettyPrintDateTime ∷ SD.TimePrecision → DT.DateTime → String
+prettyPrintDateTime prec dt =
+  prettyPrintDate (DT.date dt)
+    <> " "
```

Comment:
The parser already accepted dates with or without the `T` by the way, I just changed the printer to omit the `T`.